metals/src/main/scala/scala/meta/internal/metals/Trace.scala

```diff
@@ -26,8 +26,13 @@ object Trace {
       // the path making it difficult to tail/cat from the terminal and cmd+click from VS Code.
       // Instead, we use the `cacheDir` which has no spaces. The logs are safe to remove so
       // putting them in the "cache directory" makes more sense compared to the "config directory".
-      AbsolutePath(projectDirectories.cacheDir)
-    }.toOption
+      val cacheDir = projectDirectories.cacheDir
+      // https://github.com/scalameta/metals/issues/5590
+      // deal with issue on windows and PowerShell, which would cause us to create a null directory in the workspace
+      if (cacheDir.startsWith("null/"))
```

Comment:
Aren't windows paths encoded differently? Maybe it's best to check if the returned path is absolute?

Reply:
Ach right, it should be `\`.  Fixed!

>  Maybe it's best to check if the returned path is absolute

Not sure how to do it properly, `toAbsolutePath` will create an absolute path out of it. I would have to probably check regexes that can match all roots? I would prefer to just check for this edge case, since we haven't had any other issues with it.

Reply:
`AbsolutePath.apply` does that anyway using `Paths.get(path).isAbsolute`.

Reply:
Right, we can use that. Changed!